### ISSUE TITLE
Still fetches only 200 even with limit set to 10000

### ISSUE THREAD

**Issue by Favna** [lmmfranco/nintendo-switch-eshop#13]
As the title says, regardless of setting the limit to way higher than 200 the lib still only returns 200 results, at least when using getGamesAmerica. In the getGamesAmerica function I added some console logs to check why this could be happening, specifically after the `let accumulatedGames`

```js
console.log(limit); // outputs 10000
console.log(!hasProp(options, "limit")); // outputs false ??
console.log(filteredResponse.games.game.length + offset); // outputs 200
console.log(filteredResponse.filter.total); // outputs 596
console.log('=========================');
```

As is evident from the latter two numbers, that part of the IF statement on line 199 resolves to true, however the statement is skipped because the `!hasProp(options, "limit")` resolves to false due to the `!` swapping the condition. I tried changing this by removing the `!` but then I get thrown this error:

```
E:\UserFiles\DevProjects\ribbon\node_modules\array-unique-x\index.js:10
        var key = property ? item[property] : item;
                                 ^

TypeError: Cannot read property 'slug' of undefined
    at E:\UserFiles\DevProjects\ribbon\node_modules\array-unique-x\index.js:10:34
    at Array.filter (<anonymous>)
    at module.exports (E:\UserFiles\DevProjects\ribbon\node_modules\array-unique-x\index.js:9:18)
    at Request.request.get [as _callback] (E:\UserFiles\DevProjects\ribbon\node_modules\nintendo-switch-eshop\index.js:197:36)
    at Request.self.callback (E:\UserFiles\DevProjects\ribbon\node_modules\request\request.js:186:22)
    at emitTwo (events.js:126:13)
    at Request.emit (events.js:214:7)
    at Request.<anonymous> (E:\UserFiles\DevProjects\ribbon\node_modules\request\request.js:1163:10)
    at emitOne (events.js:116:13)
    at Request.emit (events.js:211:7)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

So basically it is useless to specify our own limit and simply flipping the condition breaks the parsing.

---

I use the lib sort of as follows (but in a Discord bot):

```js
const eshop = require('nintendo-switch-eshop'),
  fs = require('fs'),
  path = require('path');

async function run() {
  fs.writeFileSync(path.join(__dirname, '../../data/databases/eshop.json'), JSON.stringify(await eshop.getGamesAmerica({
    limit: 10000
  })), 'utf8');
}

run();
```

p.s. I know that omitting the limit altogether will have the lib fetch all possible entries, however I want to stipulate here that setting a limit far higher than the amount should also do this.

**Comment by Favna** [comment]
Ah. That explains it yes. And yeah I just checked, the JSON has a length of 596 (the current total). 

---

The initial report for my bot came in for it not having Octopath Traveler but between posting the issue and now I also manually requested the API with Postman (structuring the URL from your code) and apparently it's not even there *at all*. Ninty stop slacking smh.

![](https://favna.s-ul.eu/SzmzcoN1.png)

**Comment by Favna** [comment]
> The initial report for my bot came in for it not having Octopath Traveler (and I later checked for 20XX and even Fortnite too, same applies) but between posting the issue and now I also manually requested the API with Postman (structuring the URL from your code) and apparently it's not even there at all. Ninty stop slacking smh.

I just did some more digging and checking each of the request querystring parameters, what exactly does `shop=ncom` do? Because if I remove that parameters I *do* get games such as Octopath Traveler, 20xx and Fortnite.

**Comment by Favna** [comment]
And that's not the only one - for some reason it also applies to the aforementioned 20XX and Fortnite (of all titles). Can including `ncom` or not be moved to RequestOptions to make it optional? All said and done I'd prefer to not have it.

**Comment by lmmfranco** [comment]
Will update as soon as possible.

**Comment by lmmfranco** [comment]
@Favna the `shop` request option is now available in version 1.1.1. Refer to the README for details.
